hpat/hiframes/boxing.py

```diff
@@ -154,12 +154,21 @@ def _infer_series_list_dtype(S):
 
 
 def _infer_index_type(index):
-    # TODO: support proper inference
+    '''
+    Convertion input index type into Numba known type
+    need to return instance of the type class
+    '''
+
+    if isinstance(index, (types.NoneType, pd.RangeIndex)) or index is None:
+        return types.none
+
     if index.dtype == np.dtype('O') and len(index) > 0:
         first_val = index[0]
         if isinstance(first_val, str):
             return string_array_type
-    return types.none
+
+    numba_index_type = numpy_support.from_dtype(index.dtype)
+    return types.Array(numba_index_type, 1, 'C')
```

Comment:
But what if we deduce Array type here, but provided index is not something we can unbox to Array later?
Are there tests for TimedeltaIndex or some other non Numeric index (see https://pandas.pydata.org/pandas-docs/stable/reference/indexing.html#numeric-index)?

Reply:
` TimedeltaIndex` is disabled. All tests are passed. Please provide a test case.